hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java

```diff
@@ -634,6 +634,18 @@ public static void closeConnectionBetween(HazelcastInstance h1, HazelcastInstanc
         }
         Node n1 = getNode(h1);
         Node n2 = getNode(h2);
+
+        try {
+            // We have to flush the operations here because only after the
+            // flushing we may be sure that there are no heartbeats left in the
+            // queues which may invalidate the suspicions.
+
+            n1.getNodeEngine().getOperationService().flush();
```

Comment:
Between the `flush` and the test assertion, other HBs might end up in the queue, that could invalidate your assertions, no? I think it would be easier if we provide an example of a test depending on this.

Reply:
You are right, since there is no "synchronization" between IO and operation threads, the connection/socket may be closed, but there are still might be some pending task on IO threads. Fixing this is even more tricky and convoluted. I'm going to throw away this flushing stuff and fallback to the HB interval/timeout lowering approach.